### PR TITLE
executor: support insert ignore on duplicate update

### PR DESCRIPTION
**Pull request by jackysp**
It seems TiDB has never supported `ignore` combine with `on duplicate key update` together in `insert` statement.
PTAL @coocood

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by coocood** [comment]
@zimulala PTAL

File: executor/write.go [hunk 1]
```diff
@@ -879,7 +879,12 @@ func (e *InsertExec) exec(ctx context.Context, rows [][]types.Datum) (types.Datu
 					if err1 != nil {
 						return nil, errors.Trace(err1)
 					}
-					if _, _, _, err = e.doDupRowUpdate(h, data, row, e.OnDuplicate); err != nil {
+					_, _, _, err = e.doDupRowUpdate(h, data, row, e.OnDuplicate)
+					if kv.ErrKeyExists.Equal(err) && e.IgnoreErr {
+						e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
```
Comment (zimulala) [inline executor/write.go]: Could we add a test for this logic?

**Pull request merged**